subsys/bluetooth/host/direction.c

```diff
@@ -189,6 +200,151 @@ static int hci_df_set_adv_cte_tx_enable(struct bt_le_ext_adv *adv,
 				   buf, NULL);
 }
 
+#if defined(CONFIG_BT_DF_CONNECTIONLESS_CTE_RX)
+static int hci_df_set_cl_cte_rx_enable(struct bt_le_per_adv_sync *sync, bool enable,
+				       const struct bt_df_per_adv_sync_cte_rx_param *params)
+{
```

Comment:
With the several layers of indentation, as well as being 100 lines long, I think perhaps some of the code could be moved into intermediary static functions to significantly improve readability, e.g. the first `if (enable)` clause seems to be a validation of the parameters.

Reply:
Extracted two additional functions.